src/lib/consensus/proof_of_stake.ml

```diff
@@ -1561,8 +1532,15 @@ module Data = struct
                     , prev_min_window_density )
                next_global_slot
                ->
+              let%bind slots_per_sub_window, sub_windows_per_window =
```

Comment:
These should not be `existed` in, because then they can be anything. You should get them from the previous consensus state.

Reply:
Updated. These were also for tests.